### PR TITLE
Implement and update readme for new env var ALLOWED_METHODS

### PR DESCRIPTION
**Pull request by nitzpo**
Implement and update readme for new env var ALLOWED_METHODS that limits the methods the tool can use

### CODE REVIEW

File: src/mcp/src/main.ts [hunk 1]
```diff
@@ -35,7 +37,7 @@ server.tool(
   {
     apiType: z.enum(["graph", "azure"]).describe("Type of Microsoft API to query. Options: 'graph' for Microsoft Graph (Entra) or 'azure' for Azure Resource Management."),
     path: z.string().describe("The Azure or Graph API URL path to call (e.g. '/users', '/groups', '/subscriptions')"),
-    method: z.enum(["get", "post", "put", "patch", "delete"]).describe("HTTP method to use"),
+    method: z.enum(allowedMethods as ["get", "post", "put", "patch", "delete"]).describe(`HTTP method to use (allowed methods: ${allowedMethods.join(", ")})`),
```
[the remark this answers is not in the document]
Comment (nitzpo) [reply src/mcp/src/main.ts:40]: Its OK like this, the function knows how to handle all these types
I want the agent to not be allowed to use them

File: src/mcp/src/main.ts [hunk 2]
```diff
@@ -26,6 +26,8 @@ let graphClient: Client | null = null;
 // Check USE_GRAPH_BETA environment variable
 const useGraphBeta = process.env.USE_GRAPH_BETA !== 'false'; // Default to true unless explicitly set to 'false'
 const defaultGraphApiVersion = getDefaultGraphApiVersion();
+const defaultAllowedMethods = ["get", "post", "put", "patch", "delete"];
+const allowedMethods: string[] = process.env.ALLOWED_METHODS ? process.env.ALLOWED_METHODS.split(",") : defaultAllowedMethods;
```
[the remark this answers is not in the document]
Comment (nitzpo) [reply src/mcp/src/main.ts]: Duplicate

**Comment by darrenjrobinson** [comment]
What is the intent of this additional env var @nitzpo ?
To provide context to the MCP client on the capabilities?
At the end of the day you can only perform what your authZ allows. And that can change by adding/removing permissions (scopes) on the App Reg. So then you'd also need to update your env var (to inform the MCP Client).

**Comment by nitzpo** [comment]
Hi all! Thanks for the comments.

The idea is to be able to limit the permissions "on the fly", without creating new entities or changing your current permissions (sometimes you don't have permissions to change the permissions...)

My original idea was to create another tool "Lokka-Readonly" that will have limited permissions; and the regular tool will have full permissions. Then limiting the usage can be done easily in the IDE :)
But - I came to realize that in MS Graph, the POST method is used for both read and write! So which methods that tool will have available?? A major issue...

That brought me to the decision to implement it by selection of the http methods; so anyone can configure the available methods to their agent according to their liking and situation.

**Comment by darrenjrobinson** [comment]
How about a different approach then to list the scopes allowed. 
It is all MSGraph so limit to either .Read/Read.All or also allow .ReadWrite / ReadWrite.All ?
The scopes are what the authZ is scoped too not the protocol used for the action.

**Comment by nitzpo** [comment]
I don't think there is a reasonable way to limit access by scope at the request time; and I don't want to require changing the actual permissions

**Comment by gauthier-bedard-ssc** [comment]
Spitballing here but if there was a way to map those graph calls to PowerShell Graph SDK cmdlet, checking for the verb would be a good starting point? GET method still have the Get verb, POST splits into New, Test, etc which can better indicate what is allowed? It's getting convoluted but I think it would cover the edge cases decently. It would be fragile though.

**Comment by AKonCloud** [comment]
> Agree blocking POST will block many read apis

But leaving it as a READ ONLY is misleading, as you can create conditional access policies, compliance polices, groups, users, basically you can create almost anything and everything.

**Comment by nitzpo** [comment]
Following all the comments here
My thoughts are that its too complicated and fragile to try and match whether an endpoint action is read only or not
I think giving the user the ability to select the HTTP methods is good value for now
This could be updated in the future if/when MSFT releases some solution for this in the API itself or its preipherals

**Comment by merill** [comment]
Closing this due to the issues it can create given the many read APIs that use POST (including batch).

The right way to do this from a security perspective is through Graph permissions.
